internal/provider/resource_host_catalog_static.go

```diff
@@ -60,7 +59,54 @@ func resourceHostCatalog() *schema.Resource {
 	}
 }
 
-func setFromHostCatalogResponseMap(d *schema.ResourceData, raw map[string]interface{}) error {
+func resourceHostCatalogStatic() *schema.Resource {
+	return &schema.Resource{
+		Description: "The static host catalog resource allows you to configure a Boundary static-type host catalog. Host " +
+			"catalogs are always part of a project, so a project resource should be used inline or you " +
+			"should have the project ID in hand to successfully configure a host catalog.",
+
+		CreateContext: resourceHostCatalogStaticCreate,
+		ReadContext:   resourceHostCatalogStaticRead,
+		UpdateContext: resourceHostCatalogStaticUpdate,
+		DeleteContext: resourceHostCatalogStaticDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			IDKey: {
+				Description: "The ID of the host catalog.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			NameKey: {
+				Description: "The host catalog name. Defaults to the resource name.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			DescriptionKey: {
+				Description: "The host catalog description.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			ScopeIdKey: {
+				Description: "The scope ID in which the resource is created.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			TypeKey: {
```

Comment:
Why do we provide the option to change the type field in these TF resources which are explicitly for 1 specific type?